runtime/collect/migrate_collection.cpp

```diff
@@ -116,3 +116,20 @@ void migrate_map(void *m) {
   migrate_collection_node((void **)&impl.root);
   migrate_champ_traversal(impl.root, 0, migrate_map_leaf);
 }
+
+using treemap = rb_tree::RBTree<rng_map::Range<KElem>, KElem>;
+void migrate_treemap(treemap t) {
```

Comment:
Looking back at the definition of `RBTree`, this copy is safe because the only state is a shared pointer - would this signature be more clear as `treemap& ` perhaps?

Reply:
Given that at the call sites
https://github.com/runtimeverification/llvm-backend/blob/e75ea3cadbfaa836c6775c28f1e37158ddd01360/runtime/collect/migrate_collection.cpp#L129
and
https://github.com/runtimeverification/llvm-backend/blob/e75ea3cadbfaa836c6775c28f1e37158ddd01360/runtime/collect/migrate_collection.cpp#L130
and
https://github.com/runtimeverification/llvm-backend/blob/e75ea3cadbfaa836c6775c28f1e37158ddd01360/runtime/collect/migrate_collection.cpp#L134
the function is called with a temporary value, I would have to assign it to a local and then pass that by reference. So effectively, we wouldn't be avoiding any copies.
If you still prefer that, let me know and I can go ahead and make the change.